microsoft/fluentui-apple
# [Xcode12 Beta Builds] Only build for arm64e for iOS device builds

### Platforms Impacted

### Description of changes
- We were adding `arm64e` flavor to all iOS builds (simulator and device)
- Xcode would ignore the arm64e slice on simulator builds because it wasn't a valid architecture
- With Xcode 12 Beta 3 and Apple Silicon, that isn't true anymore, and arm64e simulator builds failed, causing our publish to fail
- Somewhat related, we were overriding ARCHS for our mac builds to add `arm64` architecture, but that's not necessary on Xcode 12 beta 3, as ARCHS_STANDARD includes arm64 in universal builds of Xcode.

The fix is to only add `arm64e` to `ARCHS` on iOS device builds and use the system standard `ARCHS` for macOS and simulator builds.

### Verification
- Locally xcode-select Xcode 12 beta 3
- run `scripts/nuget_publish.sh` from the root of the repo, ensure iphonesimulator builds fail before these changes, ensure they succeed after
- Run `lipo -archs` on the binaries afterwards to ensure the proper architectures are built for each platform

```
$ lipo -archs fluentui-apple/DerivedData/Build/Products/Debug/FluentUI.framework/Versions/A/FluentUI 
x86_64 arm64
$ lipo -archs fluentui-apple/DerivedData/Build/Products/Debug-iphoneos/libFluentUILib.a 
arm64 arm64e
$ lipo -archs fluentui-apple/DerivedData/Build/Products/Debug-iphonesimulator/libFluentUILib.a 
arm64 x86_64
```

### Pull request checklist

This PR has considered:


###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui-apple/pull/162)
